api/src/client/AwsMessagingServiceClient.ts

```diff
@@ -0,0 +1,111 @@
+import { InvokeCommand, LambdaClient } from "@aws-sdk/client-lambda";
+import { HealthCheckMetadata, MessagingServiceClient } from "@domain/types";
+import { LogWriterType } from "@libs/logWriter";
+import { ReasonPhrases } from "http-status-codes";
+
+export const AwsMessagingServiceClient = (config: {
+  functionName: string;
+  logWriter: LogWriterType;
+}): MessagingServiceClient => {
+  const logId = config.logWriter.GetId();
+  const lambdaClient = new LambdaClient({});
+
+  const sendEmail = async (
+    toEmailAddress: string,
+    options?: { templateId?: string; userId?: string },
```

Comment:
The Messaging Service will use the user's id to determine their comms preferences and also pull out their email address.

I'd pass the type of message we want to send (welcome, reminder, etc.) and then let the Messaging Service determine the template to use. Otherwise, we'll have to update every invocation of the function to account for any template changes.

```suggestion
    userId: string,
    messageType: string,
```